### ISSUE TITLE
Missing Test Case - 3008. Find Beautiful Indices in the Given Array II

### ISSUE THREAD

**Issue by A1exL1ang** [LeetCode-Feedback/LeetCode-Feedback#19363]
### Leetcode Username

Superultra

### Problem number, title, and link

3008. Find Beautiful Indices in the Given Array II https://leetcode.com/problems/find-beautiful-indices-in-the-given-array-ii

### Category of the bug
- [ ] Problem description
- [ ] Solved examples
- [ ] Problem constraints
- [ ] Problem hints
- [ ] Incorrect or missing "Related Topics"
- [ ] Incorrect test Case *(Output of test case is incorrect as per the problem statement)*
- [X] Missing test Case *(Incorrect/Inefficient Code getting accepted because of missing test cases)*
- [ ] Editorial
- [ ] Programming language support

### Description of the bug

Incorrect hashes lead to collisions and wrong answer. Specifically, there is a **common hash** that relies on overflow of unsigned long long and the following test case makes such hash get wrong answer (**regardless of the prime used**). 

I also noticed that some people are not randomizing their polynomial hash so **I made this test fail some of those submissions too** by creating strings that simultaneously fail multiple hashes (and by leveraging the fact that the problem gives us 3 strings to work with).

TLDR this one test breaks many different types of hashing solutions.

### The test (it's kinda long)
[test.txt](https://github.com/LeetCode-Feedback/LeetCode-Feedback/files/13930628/test.txt)

### C++

Below is an example of a hash that uses unsigned long long overflow (you can also access it here https://leetcode.com/contest/weekly-contest-380/submissions/detail/1145524586/).

```cpp
class Solution {
public:
	using nagai = unsigned long long;
    vector<int> beautifulIndices(string s, string a, string b, int k) {
		int p = 239;
        int n = s.size();

		vector<nagai> ha(s.size() + 1);
		for (int i = 0; i < s.size(); ++i) ha[i + 1] = ha[i] * p + s[i];
		nagai aha = 0, bha = 0;
		for (int i = 0; i < a.size(); ++i) aha = aha * p + a[i];
		for (int i = 0; i < b.size(); ++i) bha = bha * p + b[i];
		vector<nagai> pw(s.size() + 1);
		pw[0] = 1;
		for (int i = 1; i <= s.size(); ++i) pw[i] = pw[i - 1] * p;
		vector<int> occa, occb;
		for (int i = 0; i < s.size(); ++i) {
            			if (i + a.size() <= s.size() && ha[i + a.size()] - ha[i] * pw[a.size()] == aha) occa.push_back(i);
			if (i + b.size() <= s.size() && ha[i + b.size()] - ha[i] * pw[b.size()] == bha) occb.push_back(i);

		}
		int lastb = -1e9;
        vector<int> res;
		for (int i = 0; i < n; ++i) {
			if (i + a.size() <= s.size() && ha[i + a.size()] - ha[i] * pw[a.size()] == aha) {
				auto rfl = lower_bound(occb.begin(), occb.end(), i);
				int dst = 1e9;
				if (rfl != occb.end()) {
					dst = min(dst, *rfl - i);
				}
				if (rfl != occb.begin()) {
					--rfl;
					dst = min(dst, i - *rfl);
				}
				if (dst <= k) res.push_back(i);
			}
		}
		return res;
    }
};
```

### Expected behavior

The expected answer is an empty vector but the above code outputs [1024].

### Screenshots

Here is an image of the above code failing:
![image](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/91511947/d7b8d3bf-c708-4a24-a299-ae9d7aba8dd3)
Here are images of the test failing some solutions that did not randomize their hash (the two fail the exact same way so the screenshots are basically the same):
![image](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/91511947/dbfa292b-c66c-4e8a-bfc7-38c84faa14ef)
![image](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/91511947/d1ddb815-c405-4fef-8615-d21c0f216096)

### Additional context
This will definitely affect some people and the rankings of Weekly Contest 380. If you are reading this and can't find the test, it's in the 5th section (control F for "the test").

**Comment by LC-Pam** [comment]
Hello,

Thank you for reaching out to us.

Your reported issue has been promptly relayed to our dedicated team for thorough investigation. We appreciate your patience and understanding as we work to address and resolve this matter.

Rest assured, we will reach out to you as soon as we have the results and updates regarding the issue.

Once again, thank you for your support and cooperation. If you have any further questions or concerns in the meantime, please feel free to let us know.

Best regards,
LeetCode Support Team

**Comment by LC-Pam** [comment]
Thank you for your time.

We've carefully considered your feedback and thoroughly examined the problem in question. After a comprehensive evaluation, the team has decided to reject this feedback.

If you have any further questions or if there's additional feedback you'd like to share, please don't hesitate to let us know. Your input is valuable, and we appreciate your continued support!

Best regards,
The LeetCode Team
